client/packages/purchase_orders/src/DetailView/LineEdit/PurchaseOrderLineEditModal.tsx

```diff
@@ -0,0 +1,126 @@
+import { ModalMode, useDialog, useNotification } from '@common/hooks';
+import { PurchaseOrderFragment } from '../../api';
+import { DialogButton, InlineSpinner } from '@common/components';
+import { useTranslation, Box } from '@openmsupply-client/common';
+import React, { useState } from 'react';
+import { PurchaseOrderLineEdit } from './PurchaseOrderLineEdit';
+import { usePurchaseOrderLine } from '../../api/hooks/usePurchaseOrderLine';
+import { ItemWithStatsFragment } from '@openmsupply-client/system';
+import { createDraftPurchaseOrderLine } from './utils';
+interface PurchaseOrderLineEditModalProps {
+  itemId: string | null;
+  purchaseOrder: PurchaseOrderFragment;
+  mode: ModalMode | null;
+  isOpen: boolean;
+  onClose: () => void;
+}
+
+export const PurchaseOrderLineEditModal = ({
+  itemId,
+  purchaseOrder,
+  mode,
+  isOpen,
+  onClose,
+}: PurchaseOrderLineEditModalProps) => {
+  const t = useTranslation();
+  const { error } = useNotification();
+
+  const lines = purchaseOrder.lines.nodes;
+
+  const [currentItem, setCurrentItem] = useState(
+    lines.find(line => line.item.id === itemId) ?? undefined
+  );
```

Comment:
left some currentItem infrastructure in as I imagine we will use similar workflows to go through all items using next / previous functionality